File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]

        highest = 0

        for pr in prices[1:]:
            if pr >= buy:
                highest = max(highest, pr - buy)
            else:
                buy = pr
        return highest











        buy = prices[0]
        sell = -1
        highest = 0

        for price in prices[1:]:
            if price < buy:
                sell = -1
                buy = price
            else:
                sell = max(sell, price)
                highest = max(sell - buy, highest)
        return highest

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int l = 0, r = 1;
        int maxProfit = 0;
        while (r < prices.Length) {
            if (prices[l] < prices[r]) {
                maxProfit = Math.Max(maxProfit, (prices[r] - prices[l]));
                r += 1;
            } else {
                l = r;
                r = l + 1;
            }
        }
        return maxProfit;
    }
}
